internal/node/fishtail/subscriber.go

```diff
@@ -0,0 +1,107 @@
+package fishtail
+
+import (
+	"encoding/json"
+
+	"github.com/eagraf/habitat-new/core/state/node"
+	"github.com/eagraf/habitat-new/internal/node/config"
+	"github.com/eagraf/habitat-new/internal/node/controller"
+	"github.com/eagraf/habitat-new/internal/node/hdb"
+)
+
+func NewFishtailIngestSubscriber(
+	nodeConfig *config.NodeConfig,
+	nodeController controller.NodeController,
+	publisher *ATProtoEventPublisher,
+) (*hdb.IdempotentStateUpdateSubscriber, error) {
+	return hdb.NewIdempotentStateUpdateSubscriber(
+		"FishtailIngestSubscriber",
+		node.SchemaName,
+		[]hdb.IdempotentStateUpdateExecutor{
+			NewFishtailIngestExecutor(nodeConfig, nodeController, publisher),
+		},
+		NewFishtailIngestRestorer(nodeConfig, publisher),
+	)
+}
+
+type FishtailIngestExecutor struct {
+	nodeController        controller.NodeController
+	atProtoEventPublisher *ATProtoEventPublisher
+}
+
+func NewFishtailIngestExecutor(nodeConfig *config.NodeConfig, controller controller.NodeController, publisher *ATProtoEventPublisher) *FishtailIngestExecutor {
+	return &FishtailIngestExecutor{
+		nodeController:        controller,
+		atProtoEventPublisher: publisher,
+	}
+}
+
+func (f *FishtailIngestExecutor) TransitionType() string {
+	return node.TransitionStartProcess
+}
+
+func (f *FishtailIngestExecutor) ShouldExecute(update hdb.StateUpdate) (bool, error) {
+	return true, nil
+}
+
+func (f *FishtailIngestExecutor) Execute(update hdb.StateUpdate) error {
```

Comment:
I don't think this needs to be part of the whole state machine ... fishtail ingesting should be dynamic things that apps can susbcribe/unsubscribe to in the middle of the process, completely decoupled from node state.